File: API/Helpers/QueryParameters.cs
namespace API.Helpers
{
    public class QueryParameters
    {
        private const int MaxSize = 100;
        private int _size = 50;

        public int Page { get; set; } = 1;
        public int Size
        {
            get => _size;
            set
            {
                if (value <= MaxSize && value >= 0)
                {
                    _size = value;
                }
                else
                {
                    _size = MaxSize;
                }
            }
        }

        public string _sortBy = "Id";
        public string SortBy
        {
            get => _sortBy;
            set
            {
                _sortBy = value;
            }
        }

        private string _sortOrder = "asc";
        public string SortOrder
        {
            get => _sortOrder;
            set
            {
                value = value.ToLower();
                if (value == "asc" || value == "desc")
                {
                    _sortOrder = value;
                }
            }
        }
    }
}

File: client/src/store/models/QueryParameters.ts
import { IQueryParameters } from '../Interfaces';
import Dictionary from './IDictionary';


export class QueryParameters implements IQueryParameters {
  protected queryParams = new Dictionary<string | number>();
  protected urlSearchParams= URLSearchParams;

  constructor() {
    this.queryParams.add('MaxSize', 100);
    this.queryParams.add('Page', 1);
    this.queryParams.add('Size', 50);
    this.queryParams.add('SortBy', 'id');
    this.queryParams.add('SortOrder', 'asc');
  }
  public set MaxSize(value: number) {
    this.queryParams.add('MaxSize', value);
  }
  public set Page(value: number) {
    this.queryParams.add('Page', value);
  }
  public set Size(value: number) {
    this.queryParams.add('Size', value);
  }
  public set SortBy(value: string) {
    this.queryParams.add('SortBy', value);
  }

  public set SortOrder(value: string) {
    this.queryParams.add('SortOrder', value);
  }

  public QueryString(): string {
    const urlSearchParams = new URLSearchParams();
    this.queryParams.getKeys().forEach((x) => {
      urlSearchParams.append(x, this.queryParams.getItem(x).toString());
    });
    return urlSearchParams.toString();
  }
}
